src/odb/include/odb/db.h

```diff
@@ -7579,12 +7575,31 @@ class dbPowerSwitch : public dbObject
   // User Code Begin dbPowerSwitch
   static dbPowerSwitch* create(dbBlock* block, const char* name);
   static void destroy(dbPowerSwitch* ps);
-  void setInSupplyPort(const std::string& in_port);
-  void setOutSupplyPort(const std::string& out_port);
+  void addInSupplyPort(const std::string& in_port);
+  void addOutSupplyPort(const std::string& out_port);
```

Comment:
should these take dbMTerm?

Reply:
this is the logical port defined the upf. It's a string that is then later mapped to a dbMTerm

Reply:
so I double checked, and these ports are the modeled one specified while creating the power switch. These are later then mapped to dbMTerms by specifying the name of these ports and there equivalent on the dbMaster